src/frontegg-auth/src/client.rs

```diff
@@ -44,6 +59,167 @@ impl Client {
             .with(RetryTransientMiddleware::new_with_policy(retry_policy))
             .build();
 
-        Self { client }
+        let inflight_requests = Arc::new(Mutex::new(HashMap::new()));
+
+        Self {
+            client,
+            inflight_requests,
+        }
+    }
+
+    /// Makes a request to the provided URL, possibly de-duping by attaching a listener to an
+    /// already in-flight request.
+    async fn make_request<Req, Resp>(&self, url: String, req: Req) -> Result<Resp, Error>
```

Comment:
This is only used as `&url` so can be made a `&'static str`? Should avoid needing to `.to_string()` the URLs skip some allocations, and the static should make it safe in the task.

Reply:
The URL is provided to us dynamically as a command line arg in envd so it can't be a `&'static str` unfortunately 😢